upnp/src/genlib/net/http/httpreadwrite.c

```diff
@@ -445,7 +445,7 @@ int http_SendMessage(SOCKINFO *info, int *TimeOut, const char *fmt, ...)
 	/* 10 byte allocated for chunk header. */
 	char Chunk_Header[CHUNK_HEADER_SIZE];
 	size_t num_read;
-	size_t amount_to_be_read = (size_t)0;
+	off_t amount_to_be_read = 0;
 	size_t Data_Buf_Size = WEB_SERVER_BUF_SIZE;
```

Comment:
I think it would be more consistent if Data_Buf_Size became an off_t, too

Reply:
Actually no, this is a memory buffer size, it should be a size_t. Like the count argument to read(2)/write(2).

- off_t is for files: 32 or 64 bits depending on large file support (_FILE_OFFSET_BITS)
- size_t is for memory buffers (32 or 64 bits depending on machine architecture).